### ISSUE TITLE
Error thrown when trying to run classify_image.py 

### ISSUE THREAD

**Issue by kleinash** [tensorflow/models#2857]
I am beginning to think this is a bug. 

I have posted a question on SO but I don't think the answer is general. 

------------------------

### System information
- **What is the top-level directory of the model you are using**:

models/tutorials/image/imagenet ( https://github.com/tensorflow/models)

- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:

Not for this issue to be present

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:

== uname -a =====================================================
Linux workstation01 4.10.0-38-generic #42~16.04.1-Ubuntu SMP Tue Oct 10 16:32:20 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

- **TensorFlow installed from (source or binary)**:
Binary

== check pips ===================================================
numpy (1.13.3)
protobuf (3.4.0)
tensorflow-gpu (1.4.0)
tensorflow-tensorboard (0.4.0rc3)

== check for virtualenv =========================================
False
- **TensorFlow version (use command below)**:

== tensorflow import ============================================
tf.VERSION = 1.4.0
tf.GIT_VERSION = v1.4.0-rc1-11-g130a514
tf.COMPILER_VERSION = v1.4.0-rc1-11-g130a514
Sanity check: array([1], dtype=int32)

== env ==========================================================
LD_LIBRARY_PATH /usr/local/cuda-8.0/lib64:/usr/local/cuda/lib64:
DYLD_LIBRARY_PATH is unset

- **Bazel version (if compiling from source)**:

NA

- **CUDA/cuDNN version**:

== cuda libs  ===================================================
/usr/local/cuda-8.0/doc/man/man7/libcudart.7
/usr/local/cuda-8.0/doc/man/man7/libcudart.so.7
/usr/local/cuda-8.0/targets/x86_64-linux/lib/libcudart_static.a
/usr/local/cuda-8.0/targets/x86_64-linux/lib/libcudart.so.8.0.61

== env ==========================================================
LD_LIBRARY_PATH /usr/local/cuda-8.0/lib64:/usr/local/cuda/lib64:
DYLD_LIBRARY_PATH is unset

== cudnn ==========================================================
$ cat /usr/local/cuda/include/cudnn.h | grep CUDNN_MAJOR -A 2  AND $ cat /usr/local/cuda-8.0/include/cudnn.h | grep CUDNN_MAJOR -A 2
CUDNN_MAJOR 5 (It doesnt matter if this is 5.1 or 6 - I get the same error)

- **GPU model and memory**:

== nvidia-smi ===================================================
Tue Nov 21 09:49:03 2017       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 384.98                 Driver Version: 384.98                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 1070    Off  | 00000000:01:00.0  On |                  N/A |
|  0%   41C    P0    39W / 180W |    213MiB /  8110MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0       993      G   /usr/lib/xorg/Xorg                           169MiB |
|    0      1657      G   compiz                                        41MiB |
+-----------------------------------------------------------------------------+

- **Exact command to reproduce**:

$ python classify_image.py 

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem

My system was working a month ago. I updated a couple things and now I am fighting to get it working again. I think this is a bug. All I am trying to do is run the examples so that I can get back to my program which is a convolutional neural net using keras. 
To reproduce this error I just have to run the standard model. 

### Source code / logs

~/TensorFlowExamples/models/tutorials/image/imag
enet$ python classify_image.py                                      
2017-11-21 09:43:09.913398: I tensorflow/core/platform/cpu_feature_gu
ard.cc:137] Your CPU supports instructions that this TensorFlow binar
y was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA
2017-11-21 09:43:10.016280: I tensorflow/stream_executor/cuda/cuda_gp
u_executor.cc:892] successful NUMA node read from SysFS had negative 
value (-1), but there must be at least one NUMA node, so returning NU
MA node zero
2017-11-21 09:43:10.016657: I tensorflow/core/common_runtime/gpu/gpu_
device.cc:1030] Found device 0 with properties: 
name: GeForce GTX 1070 major: 6 minor: 1 memoryClockRate(GHz): 1.7465
pciBusID: 0000:01:00.0
totalMemory: 7.92GiB freeMemory: 7.62GiB
2017-11-21 09:43:10.016670: I tensorflow/core/common_runtime/gpu/gpu$device.cc:1120] Creating TensorFlow device (/device:GPU:0) -> (devic$: 0, name: GeForce GTX 1070, pci bus id: 0000:01:00.0, compute capab$lity: 6.1)
2017-11-21 09:43:10.316462: W tensorflow/core/framework/op_def_util.$c:334] Op BatchNormWithGlobalNormalization is deprecated. It will ce$se to work in GraphDef version 9. Use tf.nn.batch_normalization().
2017-11-21 09:43:10.552632: E tensorflow/stream_executor/cuda/cuda_b$as.cc:366] failed to create cublas handle: CUBLAS_STATUS_NOT_INITIAL$ZED
2017-11-21 09:43:11.119849: E tensorflow/stream_executor/cuda/cuda_d$n.cc:385] could not create cudnn handle: CUDNN_STATUS_INTERNAL_ERROR
2017-11-21 09:43:11.119875: E tensorflow/stream_executor/cuda/cuda_d$n.cc:352] could not destroy cudnn handle: CUDNN_STATUS_BAD_PARAM
2017-11-21 09:43:11.119915: F tensorflow/core/kernels/conv_ops.cc:66$] Check failed: stream->parent()->GetConvolveAlgorithms( conv_parame$ers.ShouldIncludeWinogradNonfusedAlgo<T>(), &algorithms) 
Aborted (core dumped)

== example showing tf imports correctly ===============================================
$ python
Python 2.7.12 (default, Nov 19 2016, 06:48:10) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
```
>>> import tensorflow as tf
>>> hello = tf.constant('Hello, TensorFlow!')                                 
>>> sess = tf.Session()
```
2017-11-21 10:07:35.756327: I tensorflow/core/platform/cpu_feature_guard.cc:13
7] Your CPU supports instructions that this TensorFlow binary was not compiled
 to use: SSE4.1 SSE4.2 AVX AVX2 FMA
2017-11-21 10:07:35.886090: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:892] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2017-11-21 10:07:35.886473: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1030] Found device 0 with properties: 
name: GeForce GTX 1070 major: 6 minor: 1 memoryClockRate(GHz): 1.7465
pciBusID: 0000:01:00.0
totalMemory: 7.92GiB freeMemory: 7.62GiB
2017-11-21 10:07:35.886487: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1120] Creating TensorFlow device (/device:GPU:0) -> (device: 0, name: GeForce GTX 1070, pci bus id: 0000:01:00.0, compute capability: 6.1)

`>>> print(sess.run(hello))`
Hello, TensorFlow!

UPD: This has not worked: https://github.com/fchollet/keras/issues/1538

**Comment by kleinash** [comment]
https://www.bountysource.com/issues/40693589-crash-could-not-create-cudnn-handle-when-convnets-are-used

These fixes have not worked for me.

**Comment by kleinash** [comment]
This is not much of a solution - but its the only one that worked:

Finally solved it.

CUDNN 6 doesn’t play nicely with TF. So if you have the library anywhere it crashes. Basically it’s version control not memory or paths. Had to make sure CUDNN v6 was no where to be found and TF was version 1.2 not 1.4.

Final set up:

Ubuntu 16.04. CUDA 8, CUDNN 5.1, TF 1.2. 

Installing tf version has to be done with

$ sudo apt install tensorflow-gpu==1.2

**Comment by ninception** [comment]
The suggested fixes did not work for me (Cuda 8, CudNN 6, ubuntu16.04, Tf1.4). 
But the suggestion by @kleinash worked for me. Downgraded CudNN to 5.1 and Tf to 1.2.
